src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -315,6 +325,13 @@ protected virtual void Initialize()
                     TraceWriter = new ConsoleTraceWriter(hostTraceLevel);
                 }
 
+                ConfigureLoggerFactory(settingsManager, metricsLogger);
+
+                // Use the startupLogger in this class as it is concerned with startup. The public Logger is used
+                // for all other logging after startup.
+                _startupLogger = ScriptConfig.HostConfig.LoggerFactory.CreateLogger(LogCategories.Startup);
```

Comment:
That's how ILogger works: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging#log-category

If we want categories to be meaningful, you need a new logger per category. The suggestion originally was to use one per-class (as the link above shows), but we decided that was even more confusing and harder to communicate to end-users when they look at their logs.

I'm open to suggestions -- we can tackle that in a subsequent checkin if we have a good solution. But this is how you define categories with ILogger.